src/database_builder.rs

```diff
@@ -32,38 +33,41 @@ impl DatabaseConfiguration {
         redb_builder
     }
 }
-/// Builder that allows you to create a [`Database`](crate::Database) instance via [`create`](Self::create) or [`open`](Self::open) etc. and [define](Self::define) models.
+/// Builder that allows you to create a [`Database`](crate::Database) instance via [`create`](Self::create) or [`open`](Self::open) etc.
 #[derive(Debug)]
-pub struct DatabaseBuilder {
-    database_configuration: DatabaseConfiguration,
-    models_builder: HashMap<String, ModelBuilder>,
+pub struct Builder {
+    database_configuration: Configuration,
 }
 
-impl DatabaseBuilder {
-    fn init<'a>(&'a self, database_instance: DatabaseInstance) -> Result<Database<'a>> {
+impl Builder {
+    fn init<'a>(
+        &self,
+        database_instance: DatabaseInstance,
+        models: &'a Models,
+    ) -> Result<Database<'a>> {
         let mut database = Database {
             instance: database_instance,
             primary_table_definitions: HashMap::new(),
             watchers: Arc::new(RwLock::new(watch::Watchers::new())),
             watchers_counter_id: AtomicU64::new(0),
         };
 
-        for (_, model_builder) in &self.models_builder {
+        for (_, model_builder) in models.models_builder.iter() {
             database.seed_model(&model_builder)?;
         }
 
         Ok(database)
     }
 }
 
-impl DatabaseBuilder {
+impl Builder {
     /// Similar to [redb::Builder::new()](https://docs.rs/redb/latest/redb/struct.Builder.html#method.new).
```

Comment:
not sure if refering documentation to completly different crate is good idea - url or docs in redb may change for any reason and your doc will become invalid with no notification. I'd just copy-paste the content here. The same goes for all other such comments.

Reply:
Thank you for this point. I have done it for the `new` function; however, for the other configurations, I left it as the behavior is really like a call to redb, and since I set up Renovate, the documentation will always be up to date.